bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java

```diff
@@ -720,14 +720,16 @@ public String registerConnectionAsApp(String oAutRedirectUrl)
         if (StringUtils.isEmpty(this.refreshToken)) {
             throw new ConnectionException("Error: No refresh token received");
         }
-        String usersMeResponseJson = makeRequestAndReturnString("GET",
-                "https://alexa.amazon.com/api/users/me?platform=ios&version=2.2.223830.0", null, false, null);
-        JsonUsersMeResponse usersMeResponse = parseJson(usersMeResponseJson, JsonUsersMeResponse.class);
-
-        URI uri = new URI(usersMeResponse.marketPlaceDomainName);
-        String host = uri.getHost();
-        setAmazonSite(host);
         try {
+            exhangeToken();
```

Comment:
Is this called twice (l. 733) intentionally?

Reply:
@mgeramb This question is still open.

Reply:
Yes, there is a domain switch between. I have added a comment so that is more clear

Reply:
ok, thanks!